test/e2e/plain_provisioner_test.go

```diff
@@ -177,6 +180,124 @@ var _ = Describe("plain provisioner bundle", func() {
+				crb = rbacv1.ClusterRoleBinding{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "rukpak-svr-crb",
+						Namespace: defaultSystemNamespace,
+					},
+
+					Subjects: []rbacv1.Subject{{Kind: "ServiceAccount", Name: "rukpak-svr-sa", Namespace: defaultSystemNamespace}},
+					RoleRef:  rbacv1.RoleRef{APIGroup: "rbac.authorization.k8s.io", Kind: "ClusterRole", Name: "bundle-reader"},
+				}
+
+				err = c.Create(ctx, &crb)
+				Expect(err).To(BeNil())
+				url := bundle.Status.ContentURL
+
+				// Create a Job that reada from the URL and outputs contents in the pod log
+				mounttoken := true
+				job = batchv1.Job{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "rukpak-svr-job",
+						Namespace: defaultSystemNamespace,
+					},
+					Spec: batchv1.JobSpec{
+						Template: corev1.PodTemplateSpec{
+							Spec: corev1.PodSpec{
+								Containers: []corev1.Container{
+									{
+										Name:    "rukpak-svr",
+										Image:   "curlimages/curl",
+										Command: []string{"sh", "-c", "curl -sSLk -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" -o - " + url + " | tar ztv"},
+									},
+								},
+								ServiceAccountName:           "rukpak-svr-sa",
+								RestartPolicy:                "Never",
+								AutomountServiceAccountToken: &mounttoken,
+							},
+						},
+					},
+				}
+				err = c.Create(ctx, &job)
+				Expect(err).To(BeNil())
+				Eventually(func() (bool, error) {
+					err = c.Get(ctx, types.NamespacedName{Name: "rukpak-svr-job", Namespace: defaultSystemNamespace}, &job)
+					if err != nil {
+						return false, err
+					}
+					return job.Status.CompletionTime != nil, err
+				}).Should(BeTrue())
+			})
+
+			By("reading the pod log", func() {
+				// Get Pod for the Job
+				pods := &corev1.PodList{}
+				Eventually(func() (bool, error) {
+					err := c.List(context.Background(), pods, client.MatchingLabels{"job-name": "rukpak-svr-job"})
+					if err != nil {
+						return false, err
+					}
+					return len(pods.Items) == 1, nil
+				}).Should(BeTrue())
+
+				Eventually(func() (bool, error) {
+					// Get logs of the Pod
+					pod = pods.Items[0]
+					logReader, err := kubeClient.CoreV1().Pods(defaultSystemNamespace).GetLogs(pod.Name, &corev1.PodLogOptions{}).Stream(context.Background())
+					if err != nil {
+						return false, err
+					}
+					buf := new(bytes.Buffer)
+					_, err = buf.ReadFrom(logReader)
+					Expect(err).To(BeNil())
+					return strings.Contains(buf.String(), "manifests/crds.yaml"), nil
+				}).Should(BeTrue())
+			})
+			err := c.Delete(ctx, &job)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+			err = c.Delete(ctx, &pod)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+			err = c.Delete(ctx, &crb)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
+			err = c.Delete(ctx, &sa)
+			Expect(client.IgnoreNotFound(err)).To(BeNil())
```

Comment:
I would suggest wrapping these delete statements in an Eventually() and check for `client.IgnoreNotFound(err) == true` to ensure everything is removed before ending the test.